modules/asset-manager-storage-fs/src/main/java/org/opencastproject/assetmanager/storage/impl/fs/AbstractFileSystemAssetStore.java

```diff
@@ -73,20 +73,24 @@ public void put(StoragePath storagePath, Source source) throws AssetStoreExcepti
     // working file repository. In the very few cases where the file is not in the working file repository,
     // this strategy leads to a minor overhead because the file not only gets downloaded and stored in the file system
     // but also a hard link needs to be created (or if that's not possible, a copy of the file.
-    final File origin = getFileFromWorkspace(source);
+    final File origin = getUniqueFileFromWorkspace(source);
     final File destination = createFile(storagePath, source);
     try {
       mkParent(destination);
       link(origin, destination);
     } catch (IOException e) {
       logger.error("Error while linking/copying file {} to {}: {}", origin, destination, getMessage(e));
       throw new AssetStoreException(e);
+    } finally {
+      if (origin != null) {
+        FileUtils.deleteQuietly(origin);
```

Comment:
Does this make sense in the context of multiple archive operations?  Ie, if I ingest, then archive, then process, then archive again.  That first archive is going to be hellishly expensive if all of the assets need to be redownloaded for processing.